src/routes/safe/components/Apps/AddAppForm.tsx

```diff
@@ -0,0 +1,203 @@
+import { Checkbox, Text, TextField } from '@gnosis.pm/safe-react-components'
+import memoize from 'lodash.memoize'
+import React, { useState } from 'react'
+import { FormSpy } from 'react-final-form'
+import styled from 'styled-components'
+
+import Field from 'src/components/forms/Field'
+import DebounceValidationField from 'src/components/forms/Field/DebounceValidationField'
+import GnoForm from 'src/components/forms/GnoForm'
+import { required } from 'src/components/forms/validator'
+import Img from 'src/components/layout/Img'
+import { getContentFromENS } from 'src/logic/wallets/getWeb3'
+import appsIconSvg from 'src/routes/safe/components/Transactions/TxsTable/TxType/assets/appsIcon.svg'
+import { isValid as isURLValid } from 'src/utils/url'
+
+import { getAppInfoFromUrl } from './utils'
+import { SafeApp } from './types'
+
+const APP_INFO: SafeApp = {
+  id: undefined,
+  url: '',
+  name: '',
+  iconUrl: appsIconSvg,
+  error: false,
+}
+
+const StyledText = styled(Text)`
+  margin-bottom: 19px;
+`
+
+const StyledTextFileAppName = styled(TextField)`
+  && {
+    width: 335px;
+  }
+`
+
+const AppInfo = styled.div`
+  margin: 36px 0 24px 0;
+
+  img {
+    margin-right: 10px;
+  }
+`
+
+const StyledCheckbox = styled(Checkbox)`
+  margin: 0;
+`
+
+const appExistsValidator = memoize((appList, value) => {
+  const exists = appList.find((a) => {
```

Comment:
`.some` suits better here than `.find`

also `.some` is slightly more performant 😄 
https://www.measurethat.net/Benchmarks/Show/5921/0/some-vs-find

Reply:
sure